wsagent/che-datasource-api/src/main/java/org/eclipse/che/datasource/api/JdbcConnectionFactory.java

```diff
@@ -0,0 +1,261 @@
+                if (trustStore != null) {
+                    sslSettings.setTrustStoreContent(Base64.decodeBase64(trustStore));
+                }
+            }
+
+            CheSSLSocketFactory.keystore.set(sslSettings);
+
+
+        } catch (Exception e) {
+            LOG.error("An error occured while getting keystore from Codenvy Preferences, JDBC connection will be performed without SSL", e);
+        }
+        final Connection connection = DriverManager.getConnection(getJdbcUrl(configuration), info);
+
+        return connection;
+    }
+
+    /**
+     * Builds a JDBC URL for a datasource.
+     *
+     * @param configuration the datasource configuration
+     * @return the URL
+     * @throws DatabaseDefinitionException in case the datasource configuration is incorrect
+     */
+    private String getJdbcUrl(final DatabaseConfigurationDTO configuration) throws DatabaseDefinitionException {
+        // Should we check and sanitize input values ?
+        if (configuration.getDatabaseType() == null) {
+            throw new DatabaseDefinitionException("Database type is null in " + configuration.toString());
+        }
+        switch (configuration.getDatabaseType()) {
+            case POSTGRES:
+                return getPostgresJdbcUrl(configuration);
+            case MYSQL:
+                return getMySQLJdbcUrl(configuration);
+            case ORACLE:
+                return getOracleJdbcUrl(configuration);
+            case JTDS:
+                return getJTDSJdbcUrl(configuration);
+            case NUODB:
+                return getNuoDBJdbcUrl(configuration);
+            case GOOGLECLOUDSQL:
+                return getMySQLJdbcUrl(configuration);
+            default:
+                throw new DatabaseDefinitionException("Unknown database type "
+                        + configuration.getDatabaseType()
+                        + " in "
+                        + configuration.toString());
+        }
+    }
+
+    /**
+     * Builds a JDBC URL for a PostgreSQL datasource.
+     *
+     * @param configuration the datasource configuration
+     * @return the URL
+     * @throws DatabaseDefinitionException in case the datasource configuration is incorrect
+     */
+    private String getPostgresJdbcUrl(final DatabaseConfigurationDTO configuration) {
+        String url = MessageFormat.format(URL_TEMPLATE_POSTGRES,
+                configuration.getHostName(),
+                Integer.toString(configuration.getPort()),
+                configuration.getDatabaseName());
+        return url;
+    }
+
+    /**
+     * Builds a JDBC URL for a MySQL datasource.
+     *
+     * @param configuration the datasource configuration
+     * @return the URL
+     * @throws DatabaseDefinitionException in case the datasource configuration is incorrect
+     */
+    private String getMySQLJdbcUrl(final DatabaseConfigurationDTO configuration) {
+        String url = MessageFormat.format(URL_TEMPLATE_MYSQL,
```

Comment:
I would like to admit that you're doing a great job, make an important contribution however I'm feeling that your formatter is not fully compliant to that in Che, could you please reformat the code according to this code style: https://github.com/eclipse/che-dev/tree/master/che-codestyle